packages/core/src/modules/protection.ts

```diff
@@ -2,6 +2,39 @@ import _ from "lodash";
 import { Context } from "../context";
 import { getSheetByIndex } from "../utils";
 
+export function checkProtectionCellCanEdit(
```

Comment:
Let's make this function generic to introduce read-only cells for other sheets too. This will solve #368. If non-authority sheets have the `lo` field in a cell, then it will be uneditable as well. Consider this implementation of `checkProtectionCellCanEdit` - 

```
export function checkProtectionCellCanEdit(
  ctx: Context,
  r: number,
  c: number,
  sheetId: string
) {
  const sheetFile = getSheetByIndex(ctx, sheetId);
  if (_.isNil(sheetFile)) {
    return true;
  }

  const aut = sheetFile.config ? sheetFile.config.authority : null;

  const isNotAuthority = _.isNil(aut) || _.isNil(aut.sheet) || aut.sheet === 0;

  const { data } = sheetFile;
  const cell = data?.[r]?.[c];

  if (isNotAuthority) {
    if (cell?.lo === 1) {
      return false;
    }
    return true;
  }
  if (cell?.lo === 0) {
    return true;
  }
  return false;
}
```